bot/beepboop.js

```diff
@@ -1,744 +1,955 @@
-						saveSession(serialized);
-					});
-					console.log("Successfully logged in to Instagram!");
-					resolve();
-				}
-				//Execute requests normally done post-login, reduces suspicion
-				process.nextTick(async() => await ig.simulate.postLoginFlow());
-		  	}).catch(IgLoginBadPasswordError, async() => {
-		  		//The password entered is wrong, so we ask the user for the correct one
-		  		let correctPassword = await askQuestion("\nThe password for the Instagram account '" + username + "'' is incorrect. Please type the correct password.\n");
-		  		//Save the correct one to account.json if "Remember me" is enabled
-		  		let configs = require("./configs/account.json");
-		  		if (configs["insta_password"]) {
-		  			configs["insta_password"] = correctPassword;
-		  			fs.writeFile("./configs/account.json", JSON.stringify(configs), function(err) {
-		  				if (err)
-		  					console.log("There was an error while trying to update account.json with the new password: " + err);
-		  			});
-		  		}
-		  		//Attempt to relogin
-		  		login(username, correctPassword);
-		  	}).catch(IgCheckpointError, async() => {
-		  		//-- This portion should THEORETICALLY work. I've never had my account get challenged before, so I have nothing to test it on.
-		  		//-- In the event that this does not work, please open a new issue at https://github.com/Garlicvideos/reddits-nightmare/issues/new
-		  		//Instagram wants us to prove that we are human
-		  		console.log("Human verification received from Instagram! Solving challenge...");
-				//Initiates the challenge
-				await Bluebird.try(async() => {
-			  		await ig.challenge.auto(true);
-			  		console.log(ig.state.challenge);
-			  	}).catch(IgCheckpointError, async() => {
-			  		//Call the checkpoint solving team
-			  		await solveChallenge();
-			  		/*
+  return new Promise(async function (resolve, reject) {
+    //Generates a ID for Instagram
+    ig.state.generateDevice(username);
+    //Check if the cookie exists
+    if (findSession()) {
+      //Imports the saved cookie
+      await ig.state.deserialize(loadSession());
+      if (ig.account) {
+        //Serialize session cookie (This gets invoked after every call to Instagram)
+        ig.request.end$.subscribe(async () => {
+          const serialized = await ig.state.serialize();
+          delete serialized.constants; //This deletes the version info, so you'll always use the version provided by the library
+          saveSession(serialized);
+        });
+        console.log("Successfully logged in to Instagram!");
+        resolve();
+      } else {
+        //Attempt to relogin
+        login(username, password);
+      }
+    } else {
+      //Login procedure
+      await Bluebird.try(async () => {
+        //Execute usual requests in the android app, not required but reduces suspicion from Instagram
+        // XXX await ig.simulate.preLoginFlow();
```

Comment:
Ok, so I did some testing and it looks like this line does not cause any issues.
As such, it's better to leave this line un-commented since it reduces the chances of getting detected as a bot.

Reply:
>  it looks like this line does not cause any issues.

In my case, unfortunately, it doesn't work unless I comment out those two lines.

        await ig.simulate.preLoginFlow();
        process.nextTick(async() => await ig.simulate.postLoginFlow());

`/api/v1/fbsearch/suggested_searches/?type=blended - 404 Not Found`

Tested on 4 linux servers with the same result.

This solution works for me:
https://github.com/jeancaffou/igdm/commit/39d4a6fba9d51f261389056dd9ceae2b6a4b3448

Reply:
Looks like you're right! I just duplicated the issue on a Linux server with Ubuntu LTS 20.04 installed. 

I'll have the PR merged. Thanks for your contribution :).